app/javascript/store/actions/loginActions.jsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import AjaxService from '../../services/ajax_service';
+
+export function loginUser(email, password, event) {
+    event.preventDefault();
+    let route = '/kirjaudu';
+    const auth_details = {
+        email: email,
+        password: password
+    };
+    return function(dispatch) {
+        AjaxService.post(
+            route,
+            auth_details,
+            (status, data) => {
+                dispatch({type: "SUBMIT_LOGIN", payload: data})
+            },
+            (status, err) => {
+                dispatch({type: "LOGIN_FAILED"})
+            }
+        );
+    }
+}
+
+export function handleUsernameValueChange(event) {
```

Comment:
Tälläsissä actioneissa, jotka käytännössä vaan muuttaa storen jotain arvoa (eli pelkkä dispatch) nimeäminen vois olla mahdollisimman samankaltainen kuin itse dispatchissa typellä. 

Olen myös miettinyt, että pitäis varmaan eriyttää kaikki dispatchit aina omiin funktioihinsa ja laittaa kaikki muu logiikka omiin metodeihinsa, ehkä eri hakemistoonkin. En ole itse näin tehnyt ainakaan vielä, mutta tuli mieleen jokin aika sitten.